scripts/generateDocs.js

```diff
@@ -0,0 +1,306 @@
+    if (index == -1) {
+        return;
+    }
+    var domainLink = fileFullName.substr(index);
+    var forwardSlashDomain = domainLink.replace(/\\/g, "/").replace(/\/\//g, "/");
+    if (domainRoot) {
+        return domainRoot + forwardSlashDomain;
+    }
+    return rootDocsDomain + forwardSlashDomain;
+}
+
+/**
+ * @param {string} name
+ */
+function markdownToHTMLExtension(name) {
+    // modifies the .md extension with .html
+    if (!name.endsWith(".md") && !name.match(/\.md#[a-zA-Z_0-9-]+$/)) {
+        return name
+    }
+
+    if (name.indexOf("http") != -1 && name.indexOf(rootDocsDomain) == -1) {
+        // Not in the root domain
+        return name
+    }
+
+    const i = name.lastIndexOf(".md");
+    return name.substr(0, i) + ".html" + name.substr(i + 3);
+}
+
+/**
+ * @param {string} fileFullName
+ */
+function generateNavSearchResult(fileFullName) {
+    var index = fileFullName.indexOf("docs")
+    var navigation = fileFullName.substring(index + "docs".length + 1);
+
+    if (navigation == "" || navigation == "/" || navigation == "\\") {
+        return "#";
+    }
+
+    var firstIndex = navigation.indexOf("\\");
+    if (firstIndex == -1) {
+        return "#";
+    }
+    var firstNav = navigation.substring(0, firstIndex);
+
+    var secondNavigation = navigation.substring(firstIndex + 1);
+    var secondIndex = secondNavigation.indexOf("\\");
+
+    if (secondIndex == -1) {
+        return "#" + firstNav
+    }
+    var secondNav = secondNavigation.substring(0, secondIndex)
+
+    return "#" + firstNav + "#" + secondNav
+}
+
+/**
+ * @param {string} treeViewDir
+ */
+async function processTreeViewLayer(treeViewDir) {
+    const is_specs_dir = path.basename(treeViewDir) == "specifications";
+    var dirents = await fs.readdir(treeViewDir, { encoding: 'utf-8', withFileTypes: true });
+    dirents.sort((a, b) => {
+        if (a.isDirectory() && !b.isDirectory()) { return -1; }
+        if (!a.isDirectory() && b.isDirectory()) { return 1; }
+        const aname = a.name.replace(/_/g, "zz");
+        const bname = b.name.replace(/_/g, "zz");
+        return (aname > bname) - (aname < bname);
+    });
+
+    var html = [];
+    var directories = [];
+
+    async function handleFile(filename, fullName, href) {
+        if (typeof href == 'undefined') {
+            const relativePath = relativeToRootDomain(fullName);
+            href = markdownToHTMLExtension(relativePath);
+        }
+
+        html.push('<a class="doc-outline-link" href="');
+        html.push(href);
+        html.push('">');
+        html.push('<li class="list-can-expand">');
+        if (filename in mapTable) {
+            html.push(mapTable[filename]);
+        } else {
+            const data = await fs.readFile(fullName, 'utf-8');
+            const lines = data.split(/\r?\n/);
+            const matches = lines[0].match(/^#+ +(.*)/);
+            if (matches) {
+                html.push(matches[1].replace(/`(.*)`/g, '<code>$1</code>'));
```

Comment:
I'm a bit nervous about this not handling escaped \`s correctly but I suppose this isn't the end of the world